envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/EnvoyLoadBalancingPrioritiesTest.kt

```diff
@@ -0,0 +1,188 @@
+            "envoy-control.envoy.snapshot.loadBalancing.priorities.dcPriorityProperties.dc2" to 1,
+            "envoy-control.envoy.snapshot.loadBalancing.priorities.dcPriorityProperties.dc3" to 2
+        )
+
+        @JvmField
+        @RegisterExtension
+        val consulClusters = ConsulMultiClusterExtension()
+
+        @JvmField
+        @RegisterExtension
+        val envoyControl =
+            EnvoyControlClusteredExtension(consulClusters.serverFirst, { properties }, listOf(consulClusters))
+
+        @JvmField
+        @RegisterExtension
+        val envoyControl2 =
+            EnvoyControlClusteredExtension(consulClusters.serverSecond, { properties }, listOf(consulClusters))
+
+        @JvmField
+        @RegisterExtension
+        val envoyControl3 =
+            EnvoyControlClusteredExtension(consulClusters.serverThird, { properties }, listOf(consulClusters))
+
+        @JvmField
+        @RegisterExtension
+        val envoy = EnvoyExtension(envoyControl)
+
+        @JvmField
+        @RegisterExtension
+        val serviceDC1_1 = EchoServiceExtension()
+
+        @JvmField
+        @RegisterExtension
+        val serviceDC1_2 = EchoServiceExtension()
+
+        @JvmField
+        @RegisterExtension
+        val serviceDC1_3 = EchoServiceExtension()
+
+        @JvmField
+        @RegisterExtension
+        val serviceDC1_4 = EchoServiceExtension()
+
+        @JvmField
+        @RegisterExtension
+        val serviceDC2_1 = EchoServiceExtension()
+
+        @JvmField
+        @RegisterExtension
+        val serviceDC2_2 = EchoServiceExtension()
+
+        @JvmField
+        @RegisterExtension
+        val serviceDC2_3 = EchoServiceExtension()
+
+        @JvmField
+        @RegisterExtension
+        val serviceDC2_4 = EchoServiceExtension()
+
+        @JvmField
+        @RegisterExtension
+        val serviceDC3_1 = EchoServiceExtension()
+
+        @JvmField
+        @RegisterExtension
+        val serviceDC3_2 = EchoServiceExtension()
+
+        @JvmField
+        @RegisterExtension
+        val serviceDC3_3 = EchoServiceExtension()
+
+        @JvmField
+        @RegisterExtension
+        val serviceDC3_4 = EchoServiceExtension()
+    }
+
+    @Test
+    fun `should route traffic to local dc when all instances are healthy`() {
+        consulClusters.serverFirst.registerService(serviceDC1_1)
+        consulClusters.serverSecond.registerService(serviceDC2_1)
+        consulClusters.serverThird.registerService(serviceDC3_1)
+        waitServiceOkAndFrom(serviceDC1_1)
+    }
+
+    @Test
+    fun `should route traffic to dc2 when there are no healthy instances in dc1`() {
+        val registeredId1 = consulClusters.serverFirst.registerService(serviceDC1_1)
+        consulClusters.serverSecond.registerService(serviceDC2_1)
+        consulClusters.serverThird.registerService(serviceDC3_1)
+        consulClusters.serverFirst.operations.deregisterService(registeredId1)
```

Comment:
I do not have the full context on how tests are implemented here (sorry, no time to dive in), but is it possible to mark the instance as not healthy instead of deregistering?   if we register and then deregister then there's little point in registering in the first place, no?  
anyway, I understand this likely does not impact the logic, so not blocking.

Reply:
As I see there is no possibility to mark as unhealthy, i updated tests a little bit, removed unnecessary registration and added some checks